crates/viewer/re_recording_panel/src/data.rs

```diff
@@ -0,0 +1,499 @@
+//! Data structures describing the contents of the recording panel.
+
+use std::collections::BTreeMap;
+use std::sync::Arc;
+use std::task::Poll;
+
+use ahash::HashMap;
+use itertools::Itertools as _;
+
+use re_entity_db::EntityDb;
+use re_entity_db::entity_db::EntityDbClass;
+use re_log_types::{ApplicationId, EntryId, TableId, natural_ordering};
+use re_redap_browser::{Entries, EntryInner, RedapServers};
+use re_smart_channel::SmartChannelSource;
+use re_types::archetypes::RecordingInfo;
+use re_types::components::{Name, Timestamp};
+use re_viewer_context::{DisplayMode, Item, ViewerContext};
+
+#[derive(Debug)]
+#[cfg_attr(feature = "testing", derive(serde::Serialize))]
+pub struct RecordingPanelData<'a> {
```

Comment:
can haz short doc string? no need to doc all the structs in here, but that's the entry point and it would be nice to point out what it's good for and the fact that it's a per-frame constructed thing, i.e. not something we maintain over time